### ISSUE TITLE
Run-AlPipeline do not import Test Tool Kits / Install Test apps after update

### ISSUE THREAD

**Issue by Bjarke848** [microsoft/navcontainerhelper#2848]
Hi Freddy,
We have been using Run-ALPipeline to ensure all our apps work together as an extra test step before we do a release. This has been working perfectly for over a year, but after we updated to version 4.. (I have tested with multiple versions) (It was working with version 3.0.1 which we have been using for some time). However, after updating BCContainerHelper, our tests started to fail. After debugging, we found that the script did not enter the section where it imports "Test Tool Kit" or where it "Installs Test Apps" like it used to (Nothing changed in the script in the meantime).

The script usually runs in Azure DevOps so I have made some changes to make it work locally 

```
[string] $AgentDirectory = "C:\AgentAsAService\_work\r1\a"
[string] $licenseFile = "C:\BC21\**.flf"
$artifactURL = Get-BCArtifactUrl -type OnPrem -country "W1" -version "21.2" -select Latest
$containerName = "ReleasePipelineContainerLocal"

$params = @{
    "accept_eula"     = $true
    "accept_outdated" = $true
    "auth"            = "Windows"
    "multitenant"     = $false
    "isolation"       = "process"
    "shortCuts"       = "None"
}

$DDInstall = { 
    Get-ChildItem -Path $AgentDirectory -Recurse -filter *.exe |
    Foreach-Object {
        if ( $_.Name -like "*service-components*" -Or $_.Name -like "*datadirector*") {
            $lscentral = $true
        }
        $containerpath = "c:\run\my\ISV\" + $_.Name
        Copy-FileToBcContainer -containerName $containerName -localPath $_.FullName -containerPath $containerpath
    }

    if ($lscentral -eq $true) {    
        Write-Host "Installing LS Components" -ForegroundColor Yellow
        
        Invoke-ScriptInBCContainer -containerName $containerName -scriptblock {
            & ((get-childitem -Path "c:\run\my\ISV" -Filter "*datadirector*").FullName) /VERYSILENT /SUPPRESSMSGBOXES /TASKS="srvfiles,clitools,navcli,navsrv" 
        } 
    }    
}

#New-Item -Path "$AgentDirectory\_Scripts" -Name "output" -ItemType "directory"
#Copy-Item -Path "$AgentDirectory\*.app" -Destination "$AgentDirectory\_Scripts\output\"


[Array]$zipDirectory = $AgentDirectory + "\*\output"
$NewAppFolder = $AgentDirectory + "\Apps.zip"
$NewTestFolder = $AgentDirectory + "\Test.zip"


Get-ChildItem -Path $zipDirectory -Exclude "*Test*" # | Compress-Archive -DestinationPath $NewAppFolder -Force
Get-ChildItem -Path $zipDirectory -Include "*Test*" -Recurse  #| Compress-Archive -DestinationPath $NewTestFolder -Force

Run-AlPipeline -NewBcContainer { param([hashtable]$params)
    New-BcContainer @params 
    Invoke-Command -ScriptBlock $DDInstall } `
    -containerName $containerName `
    -artifact $artifactURL `
    -baseFolder $AgentDirectory `
    -installApps @($NewAppFolder) `
    -installTestApps @($NewTestFolder) `
    -licenseFile $licenseFile `
    -enableCodeCop `
    -enableUICop `
    -enablePerTenantExtensionCop `
    -installPerformanceToolkit `
    -installTestFramework `
    -installTestLibraries `
    -installTestRunner `
    -testResultsFormat "JUnit" `
    -testResultsFile "testresults" `
    -createRuntimePackages `
    -azureDevOps `
    -assignPremiumPlan `
    -failOn "warning"
```

**Full output of scripts**
```
WARNING: No app folders found
  _____                               _                
 |  __ \                             | |               
 | |__) |_ _ _ __ __ _ _ __ ___   ___| |_ ___ _ __ ___ 
 |  ___/ _` | '__/ _` | '_ ` _ \ / _ \ __/ _ \ '__/ __|
 | |  | (_| | | | (_| | | | | | |  __/ |_  __/ |  \__ \
 |_|   \__,_|_|  \__,_|_| |_| |_|\___|\__\___|_|  |___/

Pipeline name                   
Container name                  ReleasePipelineContainerLocal
Image name                      my
ArtifactUrl                     https://bcartifacts.azureedge.net/onprem/21.2.49946.49990/w1
SasToken                        Not Specified
BcAuthContext                   Not Specified
Environment                     
ReUseContainer                  False
KeepContainer                   False
Auth                            UserPassword
Credential                      admin/Zuza3668
CompanyName                     
MemoryLimit                     8G
FailOn                          warning
TreatTestFailuresAsWarnings     False
Enable Task Scheduler           False
Assign Premium Plan             True
Install Test Runner             True
Install Test Framework          True
Install Test Libraries          True
Install Perf. Toolkit           True
InstallOnlyReferencedApps       False
generateDependencyArtifact      False
CopySymbolsFromContainer        False
enableCodeCop                   True
enableAppSourceCop              False
enableUICop                     True
enablePerTenantExtensionCop     True
doNotPerformUpgrade             False
doNotPublishApps                False
uninstallRemovedApps            False
escapeFromCops                  False
doNotBuildTests                 False
doNotRunTests                   False
doNotRunBcptTests               False
useDefaultAppSourceRuleSet      False
rulesetFile                     
azureDevOps                     True
gitLab                          False
gitHubActions                   False
License file                    Specified
CodeSignCertPfxFile             Not specified
CodeSignCertPfxPassword         Not specified
CodeSignCertIsSelfSigned        False
KeyVaultCertPfxFile             Not specified
KeyVaultCertPfxPassword         Not specified
KeyVaultClientId                
BuildOutputFile                 
ContainerEventLogFile           
TestResultsFile                 C:\AgentAsAService\_work\r1\a\testresults
BcptTestResultsFile             C:\AgentAsAService\_work\r1\a\bcptTestResults.json
TestResultsFormat               JUnit
AdditionalCountries             
PackagesFolder                  C:\AgentAsAService\_work\r1\a\.packages
OutputFolder                    C:\AgentAsAService\_work\r1\a\.output
BuildArtifactFolder             
CreateRuntimePackages           True
AppVersion                      
AppBuild                        0
AppRevision                     0
Install Apps
- C:\AgentAsAService\_work\r1\a\Apps.zip
Install Test Apps
- C:\AgentAsAService\_work\r1\a\Test.zip
Previous Apps
- None
Application folders
- None
Test application folders
- None
BCPT Test application folders
- None
Custom CodeCops
- None
NewBccontainer override
 param([hashtable]$params)
    New-BcContainer @params 
    Invoke-Command -ScriptBlock $DDInstall 

  _____       _ _ _                                          _        _                            
 |  __ \     | | (_)                                        (_)      (_)                           
 | |__) |   _| | |_ _ __   __ _    __ _  ___ _ __   ___ _ __ _  ___   _ _ __ ___   __ _  __ _  ___ 
 |  ___/ | | | | | | '_ \ / _` |  / _` |/ _ \ '_ \ / _ \ '__| |/ __| | | '_ ` _ \ / _` |/ _` |/ _ \
 | |   | |_| | | | | | | | (_| | | (_| |  __/ | | |  __/ |  | | (__  | | | | | | | (_| | (_| |  __/
 |_|    \__,_|_|_|_|_| |_|\__, |  \__, |\___|_| |_|\___|_|  |_|\___| |_|_| |_| |_|\__,_|\__, |\___|
                           __/ |   __/ |                                                 __/ |     
                          |___/   |___/                                                 |___/      

Pulling mcr.microsoft.com/businesscentral:10.0.17763.3770

Pulling generic image took 1 seconds

   _____                _   _                               _        _                 
  / ____|              | | (_)                             | |      (_)                
 | |     _ __ ___  __ _| |_ _ _ __   __ _    ___ ___  _ __ | |_ __ _ _ _ __   ___ _ __ 
 | |    | '__/ _ \/ _` | __| | '_ \ / _` |  / __/ _ \| '_ \| __/ _` | | '_ \ / _ \ '__|
 | |____| | |  __/ (_| | |_| | | | | (_| | | (__ (_) | | | | |_ (_| | | | | |  __/ |   
  \_____|_|  \___|\__,_|\__|_|_| |_|\__, |  \___\___/|_| |_|\__\__,_|_|_| |_|\___|_|   
                                     __/ |                                             
                                    |___/                                              

WARNING: Container name should not exceed 15 characters
BcContainerHelper is version 4.0.9
BcContainerHelper is running as administrator
Hyper-V is Disabled
UsePsSession is True
Host is Microsoft Windows Server 2019 Standard - ltsc2019
Docker Client Version is 20.10.6
Docker Server Version is 20.10.6
Removing Desktop shortcuts
Fetching all docker images
Fetching all docker volumes
ArtifactUrl and ImageName specified
Image my:onprem-21.2.49946.49990-w1 already exists
Using image my:onprem-21.2.49946.49990-w1
Creating Container ReleasePipelineContainerLocal
Style: onprem
Multitenant: No
Version: 21.2.49946.49990
Platform: 21.0.49939.49984
Generic Tag: 1.0.2.13
Container OS Version: 10.0.17763.3770 (ltsc2019)
Host OS Version: 10.0.17763.3770 (ltsc2019)
Using process isolation
Using locale en-US
Disabling the standard eventlog dump to container log every 2 seconds (use -dumpEventLog to enable)
Using license file C:\BC21\22.12.14_KVIK_DK_D365BC_v21_5359959.flf
Additional Parameters:
--volume "C:\AgentAsAService\_work\r1\a:c:\sources"
--env customNavSettings=EnableTaskScheduler=False
Files in C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\my:
- AdditionalOutput.ps1
- license.flf
- MainLoop.ps1
- SetupNavUsers.ps1
- SetupVariables.ps1
- updatehosts.ps1
Creating container ReleasePipelineContainerLocal from image my:onprem-21.2.49946.49990-w1
a1aa5024053936856bb38bf532aa641f85c420eda5bda70facf79e745d25e997
Waiting for container ReleasePipelineContainerLocal to be ready
Adding RELEASEPIPELINE to hosts file
Initializing...
Setting host.docker.internal to 10.1.0.21 in container hosts file (copy from host hosts file)
Setting gateway.docker.internal to 10.1.0.21 in container hosts file (copy from host hosts file)
Setting kubernetes.docker.internal to 127.0.0.1 in container hosts file (copy from host hosts file)
Setting host.containerhelper.internal to 172.24.96.1 in container hosts file
Starting Container
Hostname is ReleasePipelineContainerLocal
PublicDnsName is ReleasePipelineContainerLocal
Using NavUserPassword Authentication
Starting Local SQL Server
Starting Internet Information Server
Creating Self Signed Certificate
Self Signed Certificate Thumbprint 58B901643F212FE533FE287901EB11E4E5EDEA83
DNS identity ReleasePipelineContainerLocal
Modifying Service Tier Config File with Instance Specific Settings
Modifying Service Tier Config File with settings from environment variable
Setting EnableTaskScheduler to False
Starting Service Tier
Registering event sources
Creating DotNetCore Web Server Instance
Using application pool name: BC
Using default container name: NavWebApplicationContainer
Copy files to WWW root C:\inetpub\wwwroot\BC
Create the application pool BC
Create website: NavWebApplicationContainer without SSL
Update configuration: navsettings.json
Done Configuring Web Client
Using license file 'c:\run\my\license.flf'
Import License
Creating http download site
Setting SA Password and enabling SA
Creating admin as SQL User and add to sysadmin
Creating SUPER user
Assign Premium plan for ADMIN
Container IP Address: 172.24.105.154
Container Hostname  : ReleasePipelineContainerLocal
Container Dns Name  : ReleasePipelineContainerLocal
Web Client          : http://ReleasePipelineContainerLocal/BC/
Dev. Server         : http://ReleasePipelineContainerLocal
Dev. ServerInstance : BC
Setting ReleasePipelineContainerLocal to 172.24.105.154 in host hosts file

Files:
http://ReleasePipelineContainerLocal:8080/ALLanguage.vsix

Container Total Physical Memory is 64.0Gb
Container Free Physical Memory is 23.6Gb

Initialization took 62 seconds
Ready for connections!
Reading CustomSettings.config from ReleasePipelineContainerLocal
Creating Desktop Shortcuts for ReleasePipelineContainerLocal
Container ReleasePipelineContainerLocal successfully created

Use:
Get-BcContainerEventLog -containerName ReleasePipelineContainerLocal to retrieve a snapshot of the event log from the container
Get-BcContainerDebugInfo -containerName ReleasePipelineContainerLocal to get debug information about the container
Enter-BcContainer -containerName ReleasePipelineContainerLocal to open a PowerShell prompt inside the container
Remove-BcContainer -containerName ReleasePipelineContainerLocal to remove the container again
docker logs ReleasePipelineContainerLocal to retrieve information about URL's again
Copy C:\AgentAsAService\_work\r1\a\ls-datadirector-3-02-114-setup.exe to container ReleasePipelineContainerLocal (c:\run\my\ISV\ls-datadirector-3-02-114-setup.exe)
Installing LS Components

Creating container took 82 seconds

 _____                _       _                   _                           _                 _          
|  __ \              | |     (_)                 | |                         | |               (_)         
| |__) |___ ___  ___ | |_   ___ _ __   __ _    __| | ___ _ __   ___ _ __   __| | ___ _ __   ___ _  ___ ___ 
|  _  // _ \ __|/ _ \| \ \ / / | '_ \ / _` |  / _` |/ _ \ '_ \ / _ \ '_ \ / _` |/ _ \ '_ \ / __| |/ _ \ __|
| | \ \  __\__ \ (_) | |\ V /| | | | | (_| | | (_| |  __/ |_) |  __/ | | | (_| |  __/ | | | (__| |  __\__ \
|_|  \_\___|___/\___/|_| \_/ |_|_| |_|\__, |  \__,_|\___| .__/ \___|_| |_|\__,_|\___|_| |_|\___|_|\___|___/
                                       __/ |            | |                                                
                                      |___/             |_|                                                

Sorted App folders
External dependencies
- None
Sorted TestApp folders
- None

  _____           _        _ _ _                                     
 |_   _|         | |      | | (_)                                    
   | |  _ __  ___| |_ __ _| | |_ _ __   __ _    __ _ _ __  _ __  ___ 
   | | | '_ \/ __| __/ _` | | | | '_ \ / _` |  / _` | '_ \| '_ \/ __|
  _| |_| | | \__ \ |_ (_| | | | | | | | (_| | | (_| | |_) | |_) \__ \
 |_____|_| |_|___/\__\__,_|_|_|_|_| |_|\__, |  \__,_| .__/| .__/|___/
                                        __/ |       | |   | |        
                                       |___/        |_|   |_|        

Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\Advanced Intercompany_2.5.0.1_runtime_21.2.49946.49990.app
Synchronizing Advanced Intercompany on tenant default
Installing Advanced Intercompany on tenant default
App Advanced Intercompany_2.5.0.1_runtime_21.2.49946.49990.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\Formpipe Lasernet Connector_4.1.1.0_Runtime_Onprem_BC21.02.app
Synchronizing Lasernet Connector OnPrem on tenant default
Installing Lasernet Connector OnPrem on tenant default
App Formpipe Lasernet Connector_4.1.1.0_Runtime_Onprem_BC21.02.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-Base_0.22.6857.0.app
Synchronizing GT-Base on tenant default
Installing GT-Base on tenant default
App IT.integro Sp. z o.o._GT-Base_0.22.6857.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\ls-retail_ls-central_system_app_21-0-0-1000312.app
Synchronizing LS Central System App on tenant default
Installing LS Central System App on tenant default
App ls-retail_ls-central_system_app_21-0-0-1000312.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\LS_Retail_LS_Central-21-0-0-1000312-runtime_package.app
Synchronizing LS Central on tenant default
Installing LS Central on tenant default
App LS_Retail_LS_Central-21-0-0-1000312-runtime_package.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-LSRetailExtension_0.6.6805.0.app
Synchronizing GT-LSRetailExtension on tenant default
Installing GT-LSRetailExtension on tenant default
App IT.integro Sp. z o.o._GT-LSRetailExtension_0.6.6805.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-CET_0.7.6249.0.app
Synchronizing GT-CET on tenant default
Installing GT-CET on tenant default
App IT.integro Sp. z o.o._GT-CET_0.7.6249.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-Intercompany_0.6.6855.0.app
Synchronizing GT-Intercompany on tenant default
Installing GT-Intercompany on tenant default
App IT.integro Sp. z o.o._GT-Intercompany_0.6.6855.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-KickBack_0.0.6721.0.app
Synchronizing GT-KickBack on tenant default
Installing GT-KickBack on tenant default
App IT.integro Sp. z o.o._GT-KickBack_0.0.6721.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\Master Data Management System_5.0.1.3.app
Synchronizing Master Data Management System on tenant default
Installing Master Data Management System on tenant default
App Master Data Management System_5.0.1.3.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-MDMSExtension_0.1.6719.0.app
Synchronizing GT-MDMSExtension on tenant default
Installing GT-MDMSExtension on tenant default
App IT.integro Sp. z o.o._GT-MDMSExtension_0.1.6719.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-MyStore_0.1.6718.0.app
Synchronizing GT-MyStore on tenant default
Installing GT-MyStore on tenant default
App IT.integro Sp. z o.o._GT-MyStore_0.1.6718.0.app successfully published
Publishing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\c11aa7a3-750a-4647-805e-cf3ff5f58b06\IT.integro Sp. z o.o._GT-MigrationTool_0.11.5585.0.app
Synchronizing GT-MigrationTool on tenant default
Installing GT-MigrationTool on tenant default
App IT.integro Sp. z o.o._GT-MigrationTool_0.11.5585.0.app successfully published

Installing apps took 263 seconds
  _____                   _               _            _       
 |  __ \                 (_)             | |          | |      
 | |__) |   _ _ __  _ __  _ _ __   __ _  | |_ ___  ___| |_ ___ 
 |  _  / | | | '_ \| '_ \| | '_ \ / _` | | __/ _ \/ __| __/ __|
 | | \ \ |_| | | | | | | | | | | | (_| | | ||  __/\__ \ |_\__ \
 |_|  \_\__,_|_| |_|_| |_|_|_| |_|\__, |  \__\___||___/\__|___/
                                   __/ |                       
                                  |___/                        
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-Base-Test_0.22.6857.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-CET-Test_0.7.6249.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-Intercompany-Test_0.6.6855.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-KickBack-Test_0.0.6721.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-LSRetailExtension-Test_0.6.6805.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-MDMSExtension-Test_0.1.6719.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-MigrationTool-Test_0.11.5585.0.app
Extracting C:\Users\ADMBVI\AppData\Local\Temp\97cd9c42-8b2c-490f-b2c6-d79d2f19eaa8\IT.integro Sp. z o.o._GT-MyStore-Test_0.1.6718.0.app
Running tests for App c99a1e47-5405-461d-b49e-d366ced917cb
Connecting to http://localhost:80/BC/cs?tenant=default
Run-Tests, open page 130455
Show form 0001fd97-0000-0001-0c00-0000836bd2d2
Title: Command Line Test Tool Edit - Command Line Test Tool - DEFAULT
Setting Extension Id c99a1e47-5405-461d-b49e-d366ced917cb
Using new test-runner mechanism
at Run-Tests, C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\PsTestTool\PsTestFunctions.ps1: line 626
at <ScriptBlock>, <No file>: line 71
Validation errors occured. Error is: {
    "Severity":  0,
    "Description":  "No installed extension was found with ID \u0027c99a1e47-5405-461d-b49e-d366ced917cb\u0027.",
    "DescriptionShort":  null,
    "IsLocal":  false,
    "OriginatingControlPath":  null
}

Exception Script Stack Trace:
at Run-Tests, C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal\PsTestTool\PsTestFunctions.ps1: line 626
at <ScriptBlock>, <No file>: line 71

PowerShell Call Stack:
at Invoke-ScriptInBcContainer, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 71
at Run-TestsInBcContainer, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-TestsInNavContainer.ps1: line 373
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 704
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 2044
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 1991
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 1935
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 766
at Run-AlPipeline, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 731
at <ScriptBlock>, C:\Powershell Scripts\Run-Alpipeline.ps1: line 47

Container Free Physical Memory: 20.3Gb

Services in container ReleasePipelineContainerLocal:
- MicrosoftDynamicsNavServer$BC is Running
- MSSQL$SQLEXPRESS is Running

at Invoke-ScriptInBcContainer, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\ContainerHandling\Invoke-ScriptInNavContainer.ps1: line 112
at Run-TestsInBcContainer, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-TestsInNavContainer.ps1: line 373
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 704
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 2044
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 1991
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 1935
at <ScriptBlock>, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 766
at Run-AlPipeline, C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-AlPipeline.ps1: line 731
at <ScriptBlock>, C:\Powershell Scripts\Run-Alpipeline.ps1: line 47
Run-TestsInBcContainer Telemetry Correlation Id: bee3a6e0-3fbe-49d2-87d2-4d747ce29df5
Removing Session ReleasePipelineContainerLocal
Removing container ReleasePipelineContainerLocal
Removing entries from hosts
Removing ReleasePipelineContainerLocal from container hosts file
Removing ReleasePipelineContainerLocal-* from container hosts file
Removing Desktop shortcuts
Removing C:\ProgramData\BcContainerHelper\Extensions\ReleasePipelineContainerLocal
Run-AlPipeline Telemetry Correlation Id: ea731e25-5161-4a6c-be5d-44d9dc7afd26
Validation errors occured. Error is: {
    "Severity":  0,
    "Description":  "No installed extension was found with ID \u0027c99a1e47-5405-461d-b49e-d366ced917cb\u0027.",
    "DescriptionShort":  null,
    "IsLocal":  false,
    "OriginatingControlPath":  null
}
At C:\Users\ADMBVI\Documents\WindowsPowerShell\Modules\BcContainerHelper\4.0.9\AppHandling\Run-TestsInNavContainer.ps1:508 char:17
+                 throw $_.Exception.Message
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Validation erro...Path":  null
}:String) [], RuntimeException
    + FullyQualifiedErrorId : Validation errors occured. Error is: {
    "Severity":  0,
    "Description":  "No installed extension was found with ID \u0027c99a1e47-5405-461d-b49e-d366ced917cb\u0027.",
    "DescriptionShort":  null,
    "IsLocal":  false,
    "OriginatingControlPath":  null
}
...
```

**Screenshots**
Before the update it looked like this when we ran the script in Azure DevOps.
![image](https://user-images.githubusercontent.com/57141525/211752443-6f10062e-c836-4bd4-9765-d9925404ea64.png)

**Additional context**
- does it happen all the time?
  - Yes it does happend everytime
- did it use to work?
  - As described earlier it used to work with the old version og BCContainerHelper / Run-Alpipeline.

**Comment by freddydk** [comment]
Fixed in the latest pre-release.

**Comment by freddydk** [comment]
Shipped in 5.0.0
